### PR TITLE
🐛 Fix Compilation

### PR DESCRIPTION
**Pull request by JuaniRios**
## What?
- Make the repo compile for a release build and testing

## Why?
- Can't start fixing tests if it won't even compile

## How?
- Remove pallet-sandbox (was pretty old already and not in use)
- Remove old migration code and runtime item
- Add `FundingSuccessThreshold` (33%) to runtime
- Fix basecall filter

## Testing?
- `cargo build --release` and `cargo test` on root folder

## Anything Else?
- Benchmarks still not compiling, will take significant rewrite

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/364?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#364** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/364?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#363** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/363?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#362** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/362?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#361** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/361?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#358** <a href="https://app.graphite.dev/github/pr/Polimec/polimec-node/358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @JuaniRios and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pallets/funding/src/functions/misc.rs [hunk 1]
```diff
@@ -415,9 +415,7 @@ impl<T: Config> Pallet<T> {
 		// * Update storage *
 		ProjectsDetails::<T>::insert(project_id, project_details);
 
-		// TODO: FIX event transmition by either doing it outside function or map ProjectStatus
```
Comment (vstam1) [inline pallets/funding/src/functions/misc.rs:418]: Keep TODO until fixed?

File: pallets/funding/src/storage_migrations.rs [hunk 2]
```diff
@@ -9,151 +9,151 @@ use serde::{Deserialize, Serialize};
...
-
-	pub type MigrationToV4<T> = frame_support::migrations::VersionedMigration<
-		3,
-		4,
-		UncheckedMigrationToV4<T>,
-		crate::Pallet<T>,
-		<T as frame_system::Config>::DbWeight,
-	>;
-}
+//
```
Comment (vstam1) [inline pallets/funding/src/storage_migrations.rs:12]: Delete instead of commenting out?

File: runtimes/polimec/src/lib.rs [hunk 3]
```diff
@@ -1046,6 +1040,7 @@ impl pallet_funding::Config for Runtime {
 	type EvaluatorSlash = EvaluatorSlash;
 	type FeeBrackets = FeeBrackets;
 	type FundingCurrency = ForeignAssets;
+	type FundingSuccessThreshold = ();
```
Comment (vstam1) [inline runtimes/polimec/src/lib.rs:1043]: Might be good to actually properly set this value to 33%?

File: runtimes/polimec/src/lib.rs [hunk 4]
```diff
@@ -244,14 +244,8 @@ impl Contains<RuntimeCall> for BaseCallFilter {
 							pallet_funding::Call::end_evaluation { .. } |
 							pallet_funding::Call::evaluate { .. } |
 							pallet_funding::Call::start_auction { .. } |
-							pallet_funding::Call::root_do_auction_opening { .. } |
-							pallet_funding::Call::root_do_start_auction_closing { .. } |
 							pallet_funding::Call::bid { .. } |
-							pallet_funding::Call::root_do_end_auction { .. } |
 							pallet_funding::Call::contribute { .. } |
-							pallet_funding::Call::root_do_community_funding { .. } |
```
Comment (vstam1) [inline runtimes/polimec/src/lib.rs:252]: Add the new runtime calls to the filter, so root -> non root call instead of only removing root?

**Review by vstam1 — CHANGES_REQUESTED** [review]

**Review by lrazovic — APPROVED** [review]

**Review by vstam1 — APPROVED** [review]

**Comment by JuaniRios** [comment]
### Merge activity

* **Aug 23, 5:56 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/364).

**Pull request merged**